### DISCUSSION TITLE
How do I build a modal with a form to edit a table row?

### DISCUSSION THREAD

**Discussion by tino** [windmill-labs/windmill#2337] [Questions]
So I managed to add a button to a table row, that opens a modal, and it shows a form. The attached runnable does this:
```js
return {
  "properties": {
    "name": {
      "type": "string",
      "default": b.selectedRow?.name
    },
    "language": {
      "type": "string",
      "default": b.selectedRow?.language
    }
  },
  "required": ["name", "language"]
}
```

But now I don't have a submit button yet.

I also see a "Submit form", which apparently can connect directly to a database query to update a row. However how do I then set the current value? eg.
```sql
-- $1 language = b.selectedRow.language
-- $2 name = b.selectedRow.name
UPDATE demo VALUES ($1::TEXT, $2::TEXT) where id = $3
```
doesn't work.

**Comment by rubenfiszel** [comment]
``` 
-- $1 language = b.selectedRow.language
-- $2 name = b.selectedRow.name
UPDATE demo VALUES ($1::TEXT, $2::TEXT) where id = $3
```

The `x = y` spec will only work with static values, in your case you likely do not need default value. You also forgot to specify $3:
``` 
-- $1 language 
-- $2 name
-- $3 id
UPDATE demo VALUES ($1::TEXT, $2::TEXT) where id = $3::UUID
```
is probably what you meant. You would then do the connection to selectedRow in the runnable inputs as eval.

As for the modal, you just need to add a button to your modal that takes as input your form and run the query of your choice.

**Reply by tino** [reply]
What do you mean by "You would then do the connection to selectedRow in the runnable inputs as eval."? Because this makes for no fields to be shown:
![CleanShot 2023-09-26 at 12 29 22@2x](https://github.com/windmill-labs/windmill/assets/99524/a00dcb09-dad7-48d4-b3e2-f288956c308c)

**Reply by tino** [reply]
Adding `setValue('e', {'language': b.selectedRow.language, 'name': b.selectedRow.name})` either before or after the `openModal()` call also doesn't work.

**Reply by rubenfiszel** [reply]
Easiest is to make the form itself an eval with the proper default value although it should work with setValue. We will investigate

**Reply by rubenfiszel** [reply]
@tino after investigating, I believe  you simply used 'e' which is the button's id instead of your form's id which is what you want to set here.

**Reply by tino** [reply]
'e' is the "submit form", 'b' is the table I want to use the selectedRow values from.
